service/mapper/transitions.go

```diff
@@ -125,6 +139,12 @@ func (t *Transitions) BootstrapState(s *State) error {
 	second := tree.RootHash()
 	t.log.Info().Uint64("height", s.height).Hex("commit", second[:]).Int("registers", len(paths)).Msg("added checkpoint tree to forest")
 
+	// collect register values from bootstrap
+	t.writeRegisterValuesFromTrie(s, tree, paths)
+
+	// delete the Trie!
+	s.forest.Reset(flow.DummyStateCommitment)
```

Comment:
with your change, it looks almost like the forest field is only used by BootstrapState.  maybe remove that from State (in a different pr)